--- u-boot-2017.03.orig/arch/arm/dts/Makefile	2017-03-13 19:54:16.000000000 +0200
+++ u-boot-2017.03/arch/arm/dts/Makefile	2017-06-11 19:12:22.437617200 +0300
@@ -79,7 +79,8 @@
	armada-xp-gp.dtb			\
	armada-xp-maxbcm.dtb			\
	armada-xp-synology-ds414.dtb		\
-	armada-xp-theadorable.dtb
+	armada-xp-theadorable.dtb		\
+	armada-385-naeba.dtb

 dtb-$(CONFIG_ARCH_UNIPHIER_LD11) += \
	uniphier-ld11-ref.dtb
