diff -Nru u-boot-2017.03.orig/arch/arm/mach-mvebu/Kconfig u-boot-2017.03/arch/arm/mach-mvebu/Kconfig
--- u-boot-2017.03.orig/arch/arm/mach-mvebu/Kconfig	2017-03-13 19:54:16.000000000 +0200
+++ u-boot-2017.03/arch/arm/mach-mvebu/Kconfig	2017-06-11 19:35:06.213650998 +0300
@@ -74,6 +74,10 @@
	bool "Support ClearFog"
	select 88F6820

+config TARGET_NAEBA
+	bool "Support Naeba machine"
+	select 88F6820
+
 config TARGET_MVEBU_DB_88F3720
	bool "Support DB-88F3720 Armada 3720"
	select ARMADA_3700
@@ -125,6 +129,7 @@
	default "ds414" if TARGET_DS414
	default "maxbcm" if TARGET_MAXBCM
	default "theadorable" if TARGET_THEADORABLE
+	default "naeba" if TARGET_NAEBA

 config SYS_CONFIG_NAME
	default "clearfog" if TARGET_CLEARFOG
@@ -137,6 +142,7 @@
	default "ds414" if TARGET_DS414
	default "maxbcm" if TARGET_MAXBCM
	default "theadorable" if TARGET_THEADORABLE
+	default "naeba" if TARGET_NAEBA

 config SYS_VENDOR
	default "Marvell" if TARGET_DB_MV784MP_GP
@@ -147,6 +153,7 @@
	default "Marvell" if TARGET_MVEBU_ARMADA_8K
	default "solidrun" if TARGET_CLEARFOG
	default "Synology" if TARGET_DS414
+	default "solidrun" if TARGET_NAEBA

 config SYS_SOC
	default "mvebu"
